### ISSUE TITLE
Improve Windows support documentation

### ISSUE THREAD

**Issue by sdeleuze** [spring-projects-experimental/spring-native#1319]
- Maven parent pom should probably updated to provide support for long commend lines, see [related PR](https://github.com/spring-projects-experimental/spring-native/pull/1311/files#diff-375ce63e702a0cada157a290b8347dfa13f0b4fba4036c346746957f463cd3bcR79-R126) from @artembilan 
- Documentation should be updated accordingly
- Shaded JAR support should be tested on the CI even if running on Linux
- To be discussed with @snicoll : for start.spring.io, do we generate that in the `pom.xml` or do we just document that in `HELP.md`

**Comment by artembilan** [comment]
Here is official Native Build Tools docs about Windows support: https://graalvm.github.io/native-build-tools/latest/maven-plugin.html#long_classpath_and_shading_support.

Another Windows issue comes from Netty and its default `AddressResolverGroup` against DNS from JNDI.
See more info here: https://github.com/spring-projects/spring-framework/issues/27749.
When that one is fixed in the next Spring Framework version, the end-user still needs to add this bean into his/her project:

```
	@Bean
	ReactorNettyHttpClientMapper reactorNettyHttpClientMapper() {
		return httpClient -> httpClient.resolver(DefaultAddressResolverGroup.INSTANCE);
	}
```

That's for now.
Probably better to have a fix for DSN resolution in native images on Windows in Netty by itself.
Or even GraalVM should do something with the DSN and JNDI for Windows since everything works out-of-the-box on regular JVM or other OSs.

**Comment by jvalkeal** [comment]
I hit a same issue when I tried native builds on github actions as it makes it easy to build on windows. I suppose @artembilan workaround changes call path as I looked windows sources and NPE comes because some jdk windows native code doesn't initialise variable:

https://github.com/openjdk/jdk/blob/da75f3c4ad5bdf25167a3ed80e51f567ab3dbd01/src/java.base/windows/classes/sun/net/dns/ResolverConfigurationImpl.java#L101

Feels very much an issue in graal itself. Should we create an issue there as I didn't find anything similar from graal issues?
